### PR TITLE
Add current-user and current-practice-user models.

### PR DESCRIPTION
**Pull request by patricksrobertson**
This gives you a mechanism to retrieve/store the current_user
from Snowflakes /me endpoint and retrieve it as necessary in your
routes.

### CODE REVIEW

File: addon/routes/index.js [hunk 1]
```diff
@@ -14,12 +15,28 @@ export default Ember.Route.extend({
...
+
 
-    return me;
+  _setPracticeUser: function(currentUser, practiceId) {
+    if (practiceId) {
+      return currentUser.practiceUserByPracticeId(practiceId);
+    } else {
+      return currentUser.get('defaultPracticeUser');
+    }
+  },
```
Comment (wm) [inline addon/routes/index.js]: should it be `_getPracticeUser` since it is not setting anything?

**Comment by wm** [comment]
:octocat:

File: addon/routes/index.js (hunk 1 above)
Comment (patricksrobertson) [inline addon/routes/index.js]: Excellent point. At one point it was using `this` all over the place and decided that was poor javascriptin' and never went back.

File: addon/adapters/current-user.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import DS from 'ember-data';
+
+export default DS.RESTAdapter.extend({
+  PATH: 'me.json',
+  host: 'CHANGEME',
+  namespace: 'api/v1',
+  token: localStorage['access_token'],
```
Comment (davejachimiak) [inline addon/adapters/current-user.js]: Just a thought: Namespacing keys in `localStorage` might be advantageous: `icis_auth.access_token` for example. The chances of an app accidentally override values set from add-ons would be lower.  Our Ember apps are a ways off from that ever happening, though.

Comment (patricksrobertson) [inline addon/adapters/current-user.js]: localStorage is constrained by hostname though

Comment (patricksrobertson) [inline addon/adapters/current-user.js]: so the thing that kinda sucks is that it's hard to only access the sexy, sexy injectable services from either the addon areas or the app. My current plan is as follows:

* Addons use a conventional name for storing things in localStorage. ("access_token", "practice_id")
* Apps abstract away the access token getting/setting via injectable service -- using the 'hidden' conventions from the addons.

**Pull request merged**